### ISSUE TITLE
I got some pop noise when enable graphicEqualiser

### ISSUE THREAD

**Issue by mengguang** [andysworkshop/usb-microphone#4]
I got some pop noise when enable graphicEqualiser.   
I attach a wav file recored by audacity.    
[pop_noise.zip](https://github.com/andysworkshop/usb-microphone/files/6988496/pop_noise.zip)
If I disable the graphicEqualiser, everything is fine, although the background noise may raise a little bit.  
The hardware is almost the same as yours except the MCU is STM32F446RE.  
Could you give me some suggestion to investigate the reason ?  
BTW, In the code :   
https://github.com/andysworkshop/usb-microphone/blob/master/Core/Inc/Audio.h#L230
what the purpose of make the LSB with a random bit ?   
Thanks in advance.

**Comment by andysworkshop** [comment]
I haven't experienced the 'pop' noise myself. It might be worth posting a question on ST's own forums to see if anyone there knows what's going on.

I found out about the need to add random noise into the LSB while researching the best way to reduce 24 bit samples to 16 bit. It came from [this answer](https://stackoverflow.com/questions/28291509/convert-24-bit-signed-int-signal-to-16-bit-signed-int-signal). I tried it with and without the random noise and to my ears I could not tell the difference but since the 'right thing to do' is to add the noise then that's what I did.
